src/Hamiltonians/Product.jl

```diff
@@ -0,0 +1,92 @@
+"""
+    Product(H1::AbstractHamiltonian, H2::AbstractHamiltonian) <: AbstractHamiltonian
+
+Product of two Hamiltonians.
+"""
+struct Product{T,H1<:AbstractHamiltonian,H2<:AbstractHamiltonian} <: AbstractHamiltonian{T}
```

Comment:
Can these be any operators since there's no `starting_address` implementation?
```suggestion
struct Product{T,H1<:AbstractOperator,H2<:AbstractOperator} <: AbstractOperator{T}
```

Reply:
Yes, thanks, and `TimeEvolution` as well, for some reason I thought it had to be a Hamiltonian.